### PR TITLE
DEV-162: Added availability date for versions

### PR DESCRIPTION
**Pull request by DustinBrett**
**Changes/Notes**
- Added the ability to set/edit a date of availability for each version
- Updating can be done via the UI or the API
- API uses unix timestamp in milliseconds
- Created migration script (requires [db-migrate PR](https://github.com/ArekSredzki/electron-release-server/pull/201))
- If the version is not yet available it will not show in the UI (except in Admin) and will not be sent as a response to calls from the api

**New Version**
- Availability date can be any time after the opening of the New Version modal
- If the availability date ends up being some time between when the modal is opened and when the record is created, then the availability date is automatically set to the creation date
- The date/time picker is using the built in HTML 5 input `datetime-local`

image

**Version Management**

image

**Edit Version (Not in edit mode)**
- Date is editable from this dialog after pressing `Edit`
- Added `Make Available` button to Edit Version modal which sets the availability date to the current date/time

image

**Edit Version (In edit mode)**

image

### CODE REVIEW

**Comment by DustinBrett** [comment]
@ArekSredzki I will resolve the conflicts in this PR as well as [202](https://github.com/ArekSredzki/electron-release-server/pull/202). These 2 PR's as well as [204](https://github.com/ArekSredzki/electron-release-server/pull/204) were not done with consideration of each other because I wasn't sure which you'd want to merge and in which order. But it's not a problem and resolving these issues should be easy.

**Comment by DustinBrett** [comment]
> @ArekSredzki I will resolve the conflicts in this PR as well as [202](https://github.com/ArekSredzki/electron-release-server/pull/202). These 2 PR's as well as [204](https://github.com/ArekSredzki/electron-release-server/pull/204) were not done with consideration of each other because I wasn't sure which you'd want to merge and in which order. But it's not a problem and resolving these issues should be easy.

On that note, I've resolved this PR 1st and will wait for it to be merged because once this one is merged then there will be a few more changes to [202](https://github.com/ArekSredzki/electron-release-server/pull/202) as the `version` model will change and references to `version.name` will likely need updating in a few spots.

File: assets/js/admin/add-version-asset-modal/add-version-asset-modal-controller.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 angular.module('app.admin.add-version-asset-modal', [])
-  .controller('AddVersionAssetModalController', ['$scope', 'DataService', '$uibModalInstance', 'versionName',
-    function($scope, DataService, $uibModalInstance, versionName) {
+  .controller('AddVersionAssetModalController', ['$scope', 'DataService', '$uibModalInstance', 'version',
+    ($scope, DataService, $uibModalInstance, version) => {
```
Comment (ArekSredzki) [inline assets/js/admin/add-version-asset-modal/add-version-asset-modal-controller.js]: When I originally wrote the electron release server in 2016 I was quite careful to not use any ES6 syntax in the browser code for compatibility reasons. That said, over three years later the world is a different place and `caniuse.com` would indicate that the changes are safe to make :)

Comment (DustinBrett) [reply assets/js/admin/add-version-asset-modal/add-version-asset-modal-controller.js]: Very reasonable. If you are interested in maintaining that pre-ES6 compatibility, I can work on adding Babel to the build process or create a PR to remove any ES6 code from the front-end.

**Pull request merged**